sql/core/src/test/java/test/org/apache/spark/sql/JavaDatasetSuite.java

```diff
@@ -291,7 +291,7 @@ public void testSetOperation() {
       unioned.collectAsList());
 
     Dataset<String> subtracted = ds.except(ds2);
-    Assert.assertEquals(Arrays.asList("abc", "abc"), subtracted.collectAsList());
+    Assert.assertEquals(Arrays.asList("abc"), subtracted.collectAsList());
```

Comment:
We have to decide what we should do next? This PR is doing `Except Distinct`. 

If we want to keep the existing behavior, which is `Except All`, we need to change the external API in Dataset.

However, the current SQL interface for `Except` is wrong. We need to correct it at least.